### PR TITLE
fix(wizard): redesign wizard stepnav (add success icon and align states)

### PR DESCRIPTION
**Pull request by kevinbuhmann**
CDE-2229, VPAT-776

## PR Checklist

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] If applicable, have a visual design approval

## PR Type

Bugfix

## What is the current behavior?

- The step number was removed for steps with error status.
- Only color was used to show wizard step completed status.
- The hover and active states used incorrect text colors.
- The stepnav item had padding resulting in the button (and thus the hover, active, and selected background colors) to be surrounded by padding.

Issue Number: CDE-2229, VPAT-776

## What is the new behavior?

- The step number is not removed for steps with error status.
- An icon, in addition to color, is used to show wizard step completed status.
- The hover and active states use correct text colors.
- The stepnav item does not have padding resulting in the button (and thus the hover, active, and selected background colors) not to be surrounded by padding.

## Does this PR introduce a breaking change?

No, but some of the wizard CSS custom properties are marked to be renamed in v18.

### CODE REVIEW

File: projects/angular/src/wizard/_wizard.clarity.scss [hunk 1]
```diff
@@ -170,35 +170,32 @@
...
       &.active {
         color: wizard-variables.$clr-wizard-stepnav-text--active;
-        .clr-wizard-stepnav-link {
-          color: wizard-variables.$clr-wizard-stepnav-text--active;
-          background-color: wizard-variables.$clr-wizard-stepnav-selected-bgcolor;
-          border-radius: wizard-variables.$clr-wizard-border-radius 0 0 wizard-variables.$clr-wizard-border-radius;
+        background-color: wizard-variables.$clr-wizard-stepnav-selected-bgcolor;
+
+        &:hover {
+          background-color: wizard-variables.$clr-wizard-stepnav-selected-hover-bgcolor;
```
[the remark this answers is not in the document]
Comment (kevinbuhmann) [reply projects/angular/src/wizard/_wizard.clarity.scss:188]: I solved this a different way that doesn't require code duplication.

File: projects/angular/src/wizard/STYLES.md [hunk 2]
```diff
@@ -8,11 +8,14 @@
 | --clr-wizard-stepnav-bgcolor                   | Wizard navigation background color                                                        |
 | --clr-wizard-main-text-color                   | Wizard text color                                                                         |
 | --clr-wizard-stepnav-text                      | Wizard navigation text color                                                              |
-| --clr-wizard-stepnav-text--active              | Wizard navigation text color of the currently active element                              |
+| --clr-wizard-stepnav-text-active               | Wizard navigation item color when active                                                  |
+| --clr-wizard-stepnav-text-hover                | Wizard navigation item color when hovered                                                 |
+| --clr-wizard-stepnav-text--active              | Wizard navigation item color when selected (misnamed)                                     |
```
Comment (valentin-mladenov) [inline projects/angular/src/wizard/STYLES.md:13]: We can add deprecation for `--active` properties in favor of `selected` and add thme for removal in v18.

```suggestion
| --clr-wizard-stepnav-text--active              | Deprecated in v17 in favor of `--clr-wizard-stepnav-text-selected`                                     |
| --clr-wizard-stepnav-text-selected              | Wizard navigation item color when selected                                   |
```

File: projects/angular/src/wizard/STYLES.md [hunk 3]
```diff
@@ -8,11 +8,14 @@
...
+| --clr-wizard-stepnav-text-disabled             | Wizard navigation item color when disabled                                                |
 | --clr-wizard-title-text                        | Wizard title text color                                                                   |
 | --clr-wizard-stepnav-border-size               | Wizard navigation border size                                                             |
 | --clr-wizard-stepnav-item-border-color         | Wizard navigation border color                                                            |
-| --clr-wizard-stepnav-item-border-color--active | Wizard navigation border color for the currently active element                           |
+| --clr-wizard-stepnav-item-border-color--active | Wizard navigation item complete border color (misnamed)                                   |
```
Comment (valentin-mladenov) [inline projects/angular/src/wizard/STYLES.md:18]: We can add deprecation for --active properties in favor of selected and add thme for removal in v18.
```suggestion
| --clr-wizard-stepnav-item-border-color--active | Deprecated in v17 in favor of  `--clr-wizard-stepnav-item-border-color-selected`                                    |
| --clr-wizard-stepnav-item-border-color-selected | Wizard navigation item complete border color                                   |
```

**Review by valentin-mladenov — COMMENTED** [review]
It's better to set a proper deprecation steps for `clr-wizard-stepnav-item-border-color--active` and `clr-wizard-stepnav-text--active` properties and variables now.

Check the flow from the other CSS deprecatetions.

File: projects/angular/src/wizard/STYLES.md (hunk 2 above)
Comment (kevinbuhmann) [reply projects/angular/src/wizard/STYLES.md:13]: No, I don't want to introduce a new variable and make the code support both. I'd rather just rename it v18.

**Review by valentin-mladenov — APPROVED** [review]
LGTM

**Pull request merged**